### ISSUE TITLE
[harmony] Export and save functions on kobo Elipsa 2E do not work

### ISSUE THREAD

**Issue by sre3** [rmkit-dev/rmkit#212]
After downloading the latest build of harmony as recommended in #211, the save and export features no longer work. The save feature allow you to save the project, but nothing appears in the load menu. The export feature shows the exported dialog but nothing shows up in the /data folder. While browsing mobileread I found [this thread](https://www.mobileread.com/forums/showpost.php?p=4358967&postcount=10) and the changes to defines.h in 6efd145c136e0a1ca1925fe73f32bf555d863aff which seems like it should work? (Apologies I am not a developer other than some rudimentary python skills) 

I have two remarks - in the export there seems to be an additional slash added in the file path: *harmony//testproject-2024...png*
Additionally, while I was on the older build of harmony the export did not work, but the save *did*. 
It did seem to be using the #define SAVE_DIR "./saved_images" line, though, as I distinctively remember the export dialog showing "Saved as ./saved_images/...png" Unfortunately I don't recall the build version I was using but I may have been using an older build before 6efd145c136e0a1ca1925fe73f32bf555d863aff... Anyhow sorry to bother you again.

Images are attached below.

![image](https://github.com/rmkit-dev/rmkit/assets/74747763/de8ec052-e97b-4eb9-ab1b-bae953d96406)
![image](https://github.com/rmkit-dev/rmkit/assets/74747763/b1c7de5d-53a8-426f-b6ca-da65f7be1f5b)
![image](https://github.com/rmkit-dev/rmkit/assets/74747763/fcd7ae60-8090-4ae0-86c4-d3c7f7e72ba2)
![image](https://github.com/rmkit-dev/rmkit/assets/74747763/cc826238-dff1-4697-9bc9-71f6f1314569)
![image](https://github.com/rmkit-dev/rmkit/assets/74747763/4d5c8731-7208-47c1-a1f1-5fda79312fae)

**Comment by sre3** [comment]
I did just notice that there is an unmerged [kobo_elipsa](/rmkit-dev/rmkit/tree/kobo_elipsa) branch... I am using the build from the  master branch, not the elipsa one.

**Comment by raisjn** [comment]
thanks for reporting this! I think master is fine to use. if you are able to diff something awesome, would be glad to support (feel free to ping on discord), otherwise I'll have to find time on the weekend

it may be a matter of those directories not existing, could be something else.

**Comment by sre3** [comment]
Unfortunately I'm pretty much completely clueless when it comes to C++, so I'm not sure I will be too much help. I also have an exam on Saturday that I really should be studying for... :(

Anyway, I tried enabling dev mode and telneting in to my elipsa, but I keep getting a port 23 unreachable error. Not sure what's going on there. Don't worry about looking into it soon though, I'm just using harmony to doodle and procrastinate, so it doesn't really bother me that I can't export. 

P.S. I did wrote a little guide on using rmkit on kobo devices during a little study break: https://sre303.net/rmkit-guide-for-kobo-devices/

**Comment by sre3** [comment]
Well, telnet decided to let me in, but it looks like there's definitely nothing there. I'm still very suspicious of that extra slash in the file path - the remarkable SAVE_DIR "/home/root/harmony/saved_images" does not have that trailing slash compared to the kobo define SAVE_DIR "/opt/data/harmony/" that does...
![Screenshot_20240131_191719](https://github.com/rmkit-dev/rmkit/assets/74747763/3a083870-35d3-41d1-b73f-0ce2c67db0c7)

**Comment by sre3** [comment]
So I tried removing the trailing slash and building it myself... and it doesn't work :( The double slash is gone in the file path, but neither the export or the save works. 

![1000024039](https://github.com/rmkit-dev/rmkit/assets/74747763/ae090155-16cb-41fd-8c09-0f0df4f98cec)

**Comment by sre3** [comment]
To clarify - does not work as in there is nothing there. Anyway, I need to get back to studying, sorry!

**Comment by raisjn** [comment]
good luck on your test :-D 

I just pushed a commit (20b9a55b3b6b953bbbfefd80fa66ef38d334fee2) that should fix this, it looks like the `:` is not well liked by the mv commands for saving / exporting files. also make sure that the directory `/opt/data/harmony`exists. 

PS: nice guide for kobo, thanks for writing it! I also pushed a fix for remux so that Nickel won't get killed. I will link the guide from rmkit when I get a moment (if you don't mind)
